prov/shm/src/smr_msg.c

```diff
@@ -228,46 +73,8 @@ static ssize_t smr_recv(struct fid_ep *ep_fid, void *buf, size_t len,
 	iov.iov_base = buf;
 	iov.iov_len = len;
 
-	return smr_generic_recv(smr_get_smr_srx(ep), &iov, &desc, 1, src_addr,
```

Comment:
Since you removed srx->lock in util_srx_generic_recv. Do you need a ep lock here to match the ep lock in progress engine?